File: Python/bitcount.py
#! /usr/bin/env python
from sys import argv

_BITS = 64


def main():
    if len(argv) < 2:
        print("You must enter a number!")
        exit(1)

    parsed = _parse(argv[1])
    if parsed is None:
        print("You must enter a number!")
        exit(1)

    print(_bitcount(int(argv[1])))


def _parse(input):
    try:
        return int(input)
    except ValueError:
        return None


def _bitcount(input):
    count = 0
    for _ in range(_BITS):
        if input & 1:
            count += 1
        input = input >> 1
    return count


if __name__ == "__main__":
    main()

File: JavaScript/bitcount.js
#! /usr/bin/env node
const BITS = 64;

const main = () => {
  const raw = process.argv[2]
  if (!raw) {
    console.log('You must enter a number!');
    process.exit(1);
  }

  const arg = parseInt(raw);
  if (isNaN(arg)) {
    console.log(`${raw} is not a number!`);
    process.exit(1);
  }

  console.log(bitcount(arg));
}

const bitcount = (n) => {
  let count = 0;
  for (let i = 0; i < BITS; i++) {
    if (n & 1 === 1) {
      count++;
    }
    n = n >> 1;
  }
  return count;
};

main();
